### DISCUSSION TITLE
Chainload IPXE

### DISCUSSION THREAD

**Discussion by metashock** [ipxe/pipxe#9] [Q&A]
hi,

is it possible to chainload ipxe via network? I mean using the existing PXE firmware which is installed on the pi?

**Comment by valtzu** [comment]
Yes it is!

Here's my `dnsmasq.conf` for that:

```
enable-tftp

dhcp-userclass=set:ipxe,iPXE
dhcp-match=set:rpi-fw,60,PXEClient:Arch:00000:UNDI:002001
dhcp-match=set:rpi-uefi,60,PXEClient:Arch:00011:UNDI:003000

pxe-service=tag:rpi-fw,0,"Raspberry Pi Boot"
dhcp-boot=tag:rpi-uefi,efi/boot/bootaa64.efi
dhcp-boot=tag:ipxe,http://10.11.12.1:8888/maas/focal/
```


At `http://10.11.12.1:8888/maas/focal/` I have
```
#!ipxe

dhcp
initrd http://10.11.12.1:8888/maas/focal/boot-initrd
chain http://10.11.12.1:8888/maas/focal/boot-kernel initrd=boot-initrd root=squash:http://10.11.12.1:8888/maas/focal/squashfs ip=dhcp overlayroot=tmpfs:recurse=0 ds=nocloud-net;s=http://10.11.12.1:8888/cloud-init/
boot
```

I also needed to set these in the Makefile when building pipxe for RPi4 and I actually used netboox.xyz fork of this repo:

```
EFI_DSC		:= edk2-platforms/Platform/RaspberryPi/RPi4/RPi4.dsc
EFI_FD		:= Build/RPi4/$(EFI_BUILD)_$(EFI_TOOLCHAIN)/FV/RPI_EFI.fd
IPXE_TGT	:= bin-arm64-efi/snp.efi
```

@PeterSurda has written a lot about this around the internet so big thanks to him I was able to make it work.


EDIT: See https://github.com/valtzu/pipxe
